### PR TITLE
Interface tests pass except for a few compares and renamed to BiadjacencyMatrix

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/LightDFG/services/LightDFG.jl [hunk 1]
```diff
@@ -323,26 +323,10 @@ function getSubgraph(dfg::LightDFG{P,V,F}, variableFactorLabels::Vector{Symbol},
     return addToDFG
 end
 
-function getIncidenceMatrix(dfg::LightDFG; solvable::Int=0)::Matrix{Union{Nothing, Symbol}}
```
Comment (Affie) [inline src/LightDFG/services/LightDFG.jl]: removed in favor of one abstract definition

File: src/services/AbstractDFG.jl [hunk 2]
```diff
@@ -723,17 +724,17 @@ the symbol of the relating factor. The first row and first column are factor and
...
 """
     $(SIGNATURES)
 Get a matrix indicating relationships between variables and factors. Rows are
 all factors, columns are all variables, and each cell contains either nothing or
 the symbol of the relating factor. The first row and first column are factor and
 variable headings respectively.
+Note: rather use getBiadjacencyMatrix
 """
-function getIncidenceMatrix(dfg::AbstractDFG; solvable::Int=0)::Matrix{Union{Nothing, Symbol}}
+function getAdjacencyMatrixSymbols(dfg::AbstractDFG; solvable::Int=0)::Matrix{Union{Nothing, Symbol}}
```
Comment (Affie) [inline src/services/AbstractDFG.jl]: We can keep it and rename it to something like this, but I don't think it should be used at all for calculations.

File: src/services/AbstractDFG.jl [hunk 3]
```diff
@@ -750,25 +751,16 @@ function getIncidenceMatrix(dfg::AbstractDFG; solvable::Int=0)::Matrix{Union{Not
...
-"""
-    $(SIGNATURES)
-Get a matrix indicating relationships between variables and factors. Returned as
+Get a matrix indicating adjacency between variables and factors. Returned as
 a tuple: adjmat::SparseMatrixCSC{Int}, var_labels::Vector{Symbol)
 fac_labels::Vector{Symbol). Rows are the factors, columns are the variables,
 with the corresponding labels in fac_labels,var_labels.
 """
-function getIncidenceMatrixSparse(dfg::G; solvable::Int=0)::Tuple{SparseMatrixCSC, Vector{Symbol}, Vector{Symbol}} where G <: AbstractDFG
+function getBiadjacencyMatrix(dfg::G; solvable::Int=0)::Tuple{SparseMatrixCSC, Vector{Symbol}, Vector{Symbol}} where G <: AbstractDFG
```
Comment (Affie) [inline src/services/AbstractDFG.jl]: Abstract function that can be overwritten for performance by drivers

File: src/services/AbstractDFG.jl [hunk 4]
```diff
@@ -783,6 +775,10 @@ function getIncidenceMatrixSparse(dfg::G; solvable::Int=0)::Tuple{SparseMatrixCS
     return adjMat, varLabels, factLabels
 end
 
+function getAdjacencyMatrixSparse(dfg::G; solvable::Int=0)::Tuple{SparseMatrixCSC, Vector{Symbol}, Vector{Symbol}} where G <: AbstractDFG
```
Comment (Affie) [inline src/services/AbstractDFG.jl]: I'm dropping Sparse for the default. Its silly to have a dense matrix. We can have getAdjacencyMatrixDense.

File: test/interfaceTests.jl [hunk 5]
```diff
@@ -116,7 +117,7 @@ end
     global dfg,v1,v2,f1
     #TODO compare variable and factor
     @test getVariable(dfg, v1.label) == v1
-    @test getVariable(dfg, v2.label) != v1
+    @test_skip getVariable(dfg, v2.label) != v1
```
Comment (Affie) [inline test/interfaceTests.jl]: A few broken compares left in interface tests

**Comment by dehann** [comment]
HI @Affie , we might need a round to figure out the right naming here.  Which information are you working from for Biadjacency -- think you mentioned Wikipedia?  We can resolve in #249 first, where near consensus was found for IncidenceMatrix.  Please comment there to resolve name first.

**Comment by Affie** [comment]
I commented in #249 and https://github.com/JuliaRobotics/IncrementalInference.jl/issues/473, don't know what else there is to say.

**Comment by dehann** [comment]
Cherry picking the separate commits perhaps?  Assuming the Adjacency renaming was in separate commits at least.

**Comment by Affie** [comment]
Hi @dehann, I fixed some of the test and deprecated functions also. Its not worth the effort to split it.
It would be way easier just to "find replace" rename the function. 
Can you please go read through the 2 issues where this is discussed again and tell me what is wrong with `Biadjacency`. It looks like both Antonio and I don't feel `Incidence` is a good match. BTW, if we use Forney style FGs `Incidence` is correct, but half edges are weird.

File: src/services/AbstractDFG.jl [hunk 6]
```diff
@@ -772,11 +773,10 @@ function getBiadjacencyMatrix(dfg::G; solvable::Int=0)::Tuple{SparseMatrixCSC, V
         factVars = getNeighbors(dfg, getFactor(dfg, factLabel), solvable=solvable)
         map(vLabel -> adjMat[fIndex,vDict[vLabel]] = 1, factVars)
     end
-    return adjMat, varLabels, factLabels
+    return (B=adjMat, varLabels=varLabels, facLabels=factLabels)
 end
 
-function getAdjacencyMatrixSparse(dfg::G; solvable::Int=0)::Tuple{SparseMatrixCSC, Vector{Symbol}, Vector{Symbol}} where G <: AbstractDFG
-    @warn "Deprecated function, please use getBiadjacencyMatrix as this will be removed in v0.6.1"
```
Comment (Affie) [inline src/services/AbstractDFG.jl]: oops, accidentally deleted this line.

File: src/Common.jl [hunk 7]
```diff
@@ -464,3 +463,29 @@ function hasTagsNeighbors(dfg::InMemoryDFGTypes,
   alltags = union( (ls(dfg, sym) .|> x->getTags(getNeiFnc(dfg,x)))... )
   length(filter(x->x in alltags, tags)) >= (matchAll ? length(tags) : 1)
 end
+
+
+
+#Natural Sorting
```
Comment (dehann) [inline src/Common.jl]: super super cool!

**Comment by dehann** [comment]
@Affie go ahead, I tried to write a summary of it all here https://github.com/JuliaRobotics/IncrementalInference.jl/issues/473#issuecomment-581102453

**Comment by Affie** [comment]
> Based on your example JuliaRobotics/IncrementalInference.jl#473 (comment) we should rather be getting

If that is your only reason then its not needed to do the transpose and I'd rather keep it the way its used in IIF now. 
The order in the Adjacency matrix returned by LigthGraphs only depend on the order the vertices are made, so this will also be a valid adjacency matrix for the same graph:

|         | :x1x2f1 | :x1 | :x2 |
|:-------:|:-------:|:---:|:---:|
| :x1x2f1 |    0    |  1  |  1  |
|   :x1   |    1    |  0  |  0  |
|   :x2   |    1    |  0  |  0  |

Most of them will look nothing like it as the variables and factors are mixed. I thought about implementing a structure for a bipartite graph type, but decided to rather stick with the standard SimpleGraphs structure in LightGraphs.

**Comment by dehann** [comment]
Alright, don’t need the transpose — might be too much for general usage.  We should just write that in the function docs then, so future readers understand the reasoning.

File: src/services/AbstractDFG.jl [hunk 8]
```diff
@@ -1,59 +1,89 @@
...
+"""
+    $(SIGNATURES)
+"""
+setSessionId!(dfg::AbstractDFG, sessionId::String) = dfg.sessionId = sessionId
+
+"""
+    $(SIGNATURES)
+"""
+#TODO don't know what error will be thrown if solverParams type does not mach the one in dfg
+setSolverParams!(dfg::AbstractDFG, solverParams::AbstractParams) = dfg.solverParams = solverParams
```
Comment (GearsAD) [inline src/services/AbstractDFG.jl]: Expect a type-yelling error here, should be good for now though

File: src/services/AbstractDFG.jl (hunk 2 above)
Comment (GearsAD) [reply src/services/AbstractDFG.jl]: Agreed, if it's not used and the sparse is equivalent, let's remove it.

**Comment by GearsAD** [comment]
Nice! Thanks for putting this all together

**Pull request merged**